src/components/AvailabilityPageSingle.js

```diff
@@ -24,34 +26,71 @@ const AvailabilityPageSingle = ({ offer, error, useChangeStatus }) => {
     offer?.name?.[i18n.language] ?? offer?.name?.[offer.mainLanguage];
   const rawStatusType = offer?.status?.type;
   const rawStatusReason = offer?.status?.reason;
+  const rawBookingAvailabilityType = offer?.bookingAvailability?.type;
 
   const [type, setType] = useState('');
-  const [reason, setReason] = useState('');
+  const [reasonMainLanguage, setReasonMainLanguage] = useState('');
+  const [bookingAvailabilityType, setBookingAvailabilityType] = useState('');
 
   useEffect(() => {
     if (!rawStatusType) return;
     setType(rawStatusType);
   }, [rawStatusType]);
 
+  useEffect(() => {
+    if (!rawBookingAvailabilityType) return;
+    setBookingAvailabilityType(rawBookingAvailabilityType);
+  }, [rawBookingAvailabilityType]);
+
   useEffect(() => {
     if (type === OfferStatus.AVAILABLE) {
-      setReason('');
+      setReasonMainLanguage('');
     }
   }, [type]);
 
   useEffect(() => {
-    const newReason = offer?.status?.reason?.[i18n.language];
-    if (!rawStatusReason || !newReason) return;
-    setReason(newReason);
-  }, [rawStatusReason]);
+    const newReasonMainLanguage = rawStatusReason?.[i18n.language];
+    if (!newReasonMainLanguage) return;
+    setReasonMainLanguage(newReasonMainLanguage);
+  }, [rawStatusReason, i18n.language]);
 
-  const handleSuccessChangeStatus = () =>
-    router.push(`/${offerType}/${offerId}/preview`);
+  const handleSuccess = () => router.push(`/${offerType}/${offerId}/preview`);
 
   const changeStatusMutation = useChangeStatus({
-    onSuccess: handleSuccessChangeStatus,
+    onSuccess: handleSuccess,
   });
 
+  const createMutationPayload = () => {
+    if (offer.calendarType === CalendarType.SINGLE) {
+      return {
+        eventId: offerId,
+        subEventIds: [0],
+        subEvents: offer?.subEvent,
+        type,
+        reason: {
+          ...(offer.status.type === type && offer.status.reason),
+          [i18n.language]: reasonMainLanguage || undefined,
```

Comment:
The variable name `reasonMainLanguage` isn't correct I think. Because the reason gets saved separately for the language that is used at the moment.

Reply:
It's indeed not the main language of the offer, maybe `reasonUserLanguage` is a better name?

Reply:
I mean, it is just the reasing you're saving in state. With those variable names it looks like it is a language like `nl` or `fr` that you're saving. I would revert it to `reason` or if that is not clear to you change it to something like `reasonInCurrentLanguage`

Reply:
Ah yes, good point. I will change it to `reasonInCurrentLanguage`